docs/docs/unreachable.md

```diff
@@ -0,0 +1,90 @@
+
+## Apache Mesos Unreachable Strategies
+
+Apache Meso's ability to communicate a task / node is unreachable is controlled by 2 concepts; 1) mesos-agent health check and 2) node rate limiter. Regarding agent health checks, the mesos-master flags of control are:
+`-max_agent_ping_timeouts` and `-agent_ping_timeout`. While the Mesos defaults are 5 and 15s respectively providing a 75 second notification event by default (assuming the loss of 1 agent). The default for DC/OS for [max_slave_ping_timeouts is 20](https://github.com/dcos/dcos/blob/9cc6ab28060545cd203c09aa7fa1b9456773d080/gen/dcos-config.yaml#L449)
+providing a 5 minute delay window between a lost node and notification to frameworks that the task is unreachable.
+
+Additionally Apache Mesos has a node rate limiter which was established prior to partition aware frameworks and is still enforced. The purpose of the rate limiter is to reduce the number of reported lost nodes / agents within a specified amount of time through the `--agent_removal_rate_limit=VALUE` flag. The default is None, which has the effect of reporting agents immediately after the health check defined above. The default in DC/OS however is [1/20mins](https://github.com/dcos/dcos/blob/9cc6ab28060545cd203c09aa7fa1b9456773d080/gen/dcos-config.yaml#L442)
+which is 1 node every 20 mins. With this default, the loss of 2 nodes would result in the first node reporting 5 mins after health checks start to fail that it is unreachable, followed by the 2nd node reporting unreachable 20 mins later. In this scenario, agents and tasks will be reported by Mesos as active for the time between the start of health check failure and the unreachable event as defined here.
+
+The net result is that the amount of time that a node on DC/OS can be lost or unreachable before Marathon is notified is not predictable. It is defined by the number of lost nodes and the order in which they are lost. This was confirmed in a test where a 5 private agent cluster had a task on 1 node. The agent processes on each node were shutdown with the node hosting the task being shutdown last. The amount of time lapsed prior to Marathon receiving a TASK_UNREACHABLE status update was > 1.25 hrs. It is important to understand that these tasks are listed by Mesos as Active until this time and Marathon is completely unaware until notified by Mesos.
+
+**NOTE:**
+There are 2 JIRAs against Mesos to remove rate limits. One marked as a critical bug https://issues.apache.org/jira/browse/MESOS-7721
+the other as a major improvement. https://issues.apache.org/jira/browse/MESOS-5948. It seems as though the default Mesos behavior is as if there is no rate limit, which begs the question why we don't set DC/OS to not have this rate limit?
```

Comment:
...so why do we have it? :)

Reply:
@jeid64 has brought up the point that there could be issues upgrading clusters if this feature is not enabled.

Which, to me, seems to indicate that we need a better mechanism to guard against agent removal during upgrades, not one that makes things worse during the actual cluster operation.